scripts/scil_split_image.py

```diff
@@ -0,0 +1,111 @@
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_dwi',
+                   help='The DW image file to split.')
+
+    p.add_argument('in_bval',
+                   help='The b-values file in FSL format (.bval).')
+
+    p.add_argument('in_bvec',
+                   help='The b-vectors file in FSL format (.bvec).')
+
+    p.add_argument('split_indices', nargs='+', type=int,
+                   help='The list of indices where to split the image. For '
+                        'example 3 10. This would split the image in three '
+                        'parts, such as [:3], [3:10], [10:]. Indices must be '
+                        'in increasing order.')
+
+    p.add_argument('--out_basename',
+                   help='The basename of the output files. Indices number '
+                        'will be appended to out_basename, such as '
+                        'out_basename_0_3, out_basename_3_10, '
+                        'out_basename_10_end. By default, indices number'
+                        'will be appended to in_dwi, '
+                        'such as in_dwi_0_3, in_dwi_3_10, in_dwi_10_end.')
+
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    if args.verbose:
+        logging.basicConfig(level=logging.INFO)
+
+    assert_inputs_exist(parser, [args.in_dwi, args.in_bval, args.in_bvec])
+    assert_outputs_exist(parser, args, [],
+                         optional=args.out_basename)
+
+    bvals, bvecs = read_bvals_bvecs(args.in_bval, args.in_bvec)
+
+    img = nib.load(args.in_dwi)
+
+    # Check if the indices fit inside the range of possible values
+    if np.max(args.split_indices) > img.shape[-1]:
+        parser.error('split_indices values must be lower than the total '
+                     'number of direcitons.')
+    if np.min(args.split_indices) <= 0:
+        parser.error('split_indices values must be higher than 0.')
+    # Check if the indices are in increasing order
+    if not np.all(np.diff(args.split_indices) > 0):
+        parser.error('split_indices values must be in increasing order.')
+
+    indices = np.concatenate(([0], args.split_indices, [img.shape[-1]]))
+
+    for i in range(len(indices)-1):
+        data_split = img.dataobj[..., indices[i]:indices[i+1]]
+        bvals_split = bvals[indices[i]:indices[i+1]]
+        bvecs_split = bvecs[indices[i]:indices[i+1]]
+        # Saving the output files
+        index_name = "_" + str(indices[i]) + "_" + str(indices[i+1])
+        if args.out_basename:
+            data_name = args.out_basename + index_name
+            bval_name = args.out_basename + index_name
+            bvec_name = args.out_basename + index_name
+        else:
+            data_name = str(Path(Path(args.in_dwi).stem).stem) + index_name
```

Comment:
Use os.path.basename(args.in_dwi).split('.')[0] or out_basename.
It seems more elegant.